### PR TITLE
🌱 Drop internal log package & improve logs and errors

### LINKED ISSUE #7150: Ensure correct logging of reference structs in both json and text logging
With recent logging changes a [reference and object logger were added to CAPI](https://github.com/kubernetes-sigs/cluster-api/commit/1a286fdf7da314bdfbb76b042f8f25f1828734a0#diff-3a86c514d2d3f52dc7235e0444e3a2c54960795d7e73cecb131b876ce306442b):

These objects may look correct in json logging, but may not correctly represent the object in text form.

https://github.com/kubernetes-sigs/cluster-api/pull/6918 Added a new logging struct `logRef` which correctly logs to both text and json formats.

We should investigate using a single structure across the codebase where references like this are used, and decide to include a single version in our internal log package, with or without helper functions.

Current reference implementation:
https://github.com/kubernetes-sigs/cluster-api/blob/1a286fdf7da314bdfbb76b042f8f25f1828734a0/internal/log/log.go#L103-L114

New reference implementation:
https://github.com/kubernetes-sigs/cluster-api/blob/a99bcf5460d83abd3201ad4441b5ce4d2529a006/test/extension/handlers/topologymutation/lib.go#L103-L127

### PR DESCRIPTION
**Pull request by sbueringer**
Signed-off-by: Stefan Büringer buringerst@vmware.com

**What this PR does / why we need it**:
Drop the internal/log package in favor of regular log statements.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #7150

### CODE REVIEW

**Comment by sbueringer** [comment]
@chrischdi I'll need some more time for a final review to ensure the log statements all make sense

**Comment by sbueringer** [comment]
/assign @chrischdi @fabriziopandini 

Ready for review :)

Would be great to get this merged soon to avoid rebases (already did a bunch of verification / self-review to avoid mistakes , don't want to repeat it)

**Review by fabriziopandini — COMMENTED** [review]
Nice cleanup, great to be back to having only one way of doing logs!
/lgtm

**Review by chrischdi — COMMENTED** [review]
Only cosmetical / optional nit on prod code and nit on `_test.go`. Feel free to do it in a follow-up PR if it makes it easier for you!

**Comment by chrischdi** [comment]
/approve
/hold

for your convenience

File: internal/controllers/topology/cluster/patches/patch.go [hunk 1]
```diff
@@ -108,7 +110,7 @@ func patchUnstructured(ctx context.Context, original, modified *unstructured.Uns
 	}
 
 	// Log the delta between the object before and after applying the accumulated patches.
-	log.V(4).WithObject(original).Infof("Applying accumulated patches to desired state: %s", string(diff))
+	log.V(4).WithValues(original.GetKind(), klog.KObj(original)).Info(fmt.Sprintf("Applying accumulated patches to desired state of %s", original.GetKind()), "patch", string(diff))
```
[the remark this answers is not in the document]
Comment (sbueringer) [reply internal/controllers/topology/cluster/patches/patch.go:113]: No reason, changed

**Comment by sbueringer** [comment]
Thx for the quick review! Should be fixed

**Pull request merged**